test/internal_rules.jl

```diff
@@ -409,6 +409,24 @@ end
                 test_reverse(f!, Const, (Y, TY), (M, TM), (B, TB), (_A, Const))
             end
         end
+        @testset "regression test for #1306" begin
```

Comment:
Can you give this a nicer name?

Reply:
Done. I also added two lines of explanation. Let me know if you need more.